src/charm.py

```diff
@@ -163,7 +201,7 @@ def _on_kafka_pebble_ready(self, event: PebbleReadyEvent) -> None:
             self.unit.status = BlockedStatus("kafka unit not connected to ZooKeeper")
             return
 
-    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+    def _on_config_changed(self, event: EventBase) -> None:
```

Comment:
Uhm, interesting. Why we want to be less specific? I have seen that in VM charms we use "EventBase", but why not ConfigChangedEvent?

Reply:
We manually call `on_config_changed` from other methods, e.g `on_storage_detaching` which has type `StorageDetachingEvent`. 
Using the generic base event type avoids Pyright typehinting warnings when calling `self._on_config_changed(event=event)`.

E.g:
![image](https://user-images.githubusercontent.com/51744472/210455985-878f47ac-ba60-428b-a83b-74fb9eb54a2b.png)

In general, we are not just calling the function implicitly from the observed events (in which case, `ConfigChangedEvent` is appropriate), because we probably shouldn't morph an event to a different type. Our `_on_config_changed()` function is defined by accepting any event type. We're doing the same in the Kafka VM too - https://github.com/canonical/kafka-operator/blob/badf25fa44f5b0483cd3a133ab072e02a322f25f/src/charm.py#L215

Reply:
I see! Makes sense!